openshift.py

```diff
@@ -0,0 +1,750 @@
+                    user not in role_binding["userNames"]
+                ):
+                    return Response(
+                        response=json.dumps(
+                            {
+                                "msg": "rolebinding does not exist - unable to delete ("
+                                + user
+                                + ","
+                                + project_name
+                                + ","
+                                + role
+                                + ")"
+                            }
+                        ),
+                        status=400,
+                        mimetype="application/json",
+                    )
+                role_binding["userNames"].remove(user)
+            else:
+                return Response(
+                    response=json.dumps(
+                        {
+                            "msg": "Invalid request ("
+                            + user
+                            + ","
+                            + project_name
+                            + ","
+                            + role
+                            + ","
+                            + op
+                            + ")"
+                        }
+                    ),
+                    status=400,
+                    mimetype="application/json",
+                )
+
+            # now add or remove the user
+            r = self.update_rolebindings(project_name, openshift_role, role_binding)
+
+            msg = "unknown message"
+            if r.status_code == 200 or r.status_code == 201:
+                if op == "add":
+                    msg = "Added role to user on project"
+                elif op == "del":
+                    msg = "removed role from user on project"
+                return Response(
+                    response=json.dumps({"msg": msg}),
+                    status=200,
+                    mimetype="application/json",
+                )
+            if op == "add":
+                msg = "unable to add role to user on project"
+            elif op == "del":
+                msg = "unable to remove role from user on project"
+            return Response(
+                response=json.dumps({"msg": msg}),
+                status=400,
+                mimetype="application/json",
+            )
+        return Response(
+            response=json.dumps(
+                {
+                    "msg": "rolebinding already exists ("
+                    + user
+                    + ","
+                    + project_name
+                    + ","
+                    + role
+                    + ")"
+                }
+            ),
+            status=400,
+            mimetype="application/json",
+        )
+
+
+class openshift_3_x(openshift):
+
+    # member functions for projects
+    def project_exists(self, project_name):
+        url = "https://" + self.get_url() + "/oapi/v1/projects/" + project_name
```

Comment:
Instead of always appending `https://` to the URL, why not just pass in an actual URL in `OPENSHIFT_URL`?